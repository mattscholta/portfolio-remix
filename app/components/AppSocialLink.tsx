import type { Social } from "~/data/resume";

export interface AppSocialLinkProps {
  data: Social;
}

export const AppSocialLink = (props: AppSocialLinkProps) => {
  const { data } = props;

  // Setup
  const alt = `Follow me on ${data.title}`;
  const size = 14;

  return (
    <a
      className="flex items-center gap-4 break-words break-all text-sm text-color-copy-light"
      href={data.url}
      key={data.title}
      rel="noopener noreferrer"
      target="_blank"
    >
      <img alt={alt} height={size} src={data.icon} width={size} />
      {data.title}
    </a>
  );
};
