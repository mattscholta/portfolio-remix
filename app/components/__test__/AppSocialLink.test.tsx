import { expect, test } from "vitest";
import { render } from "@testing-library/react";

import { AppSocialLink } from "../AppSocialLink";

describe("AppSocialLink Component", () => {
  const component = render(
    <AppSocialLink
      data={{
        icon: "/path/to/facebook.svg",
        title: "Facebook",
        url: "https://facebook.com"
      }}
    />
  );

  test("renders the facebook icon and link", () => {
    expect(component.baseElement).toMatchInlineSnapshot(`
      <body>
        <div>
          <a
            class="flex items-center gap-4 break-words break-all text-sm text-color-copy-light"
            href="https://facebook.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="Follow me on Facebook"
              height="14"
              src="/path/to/facebook.svg"
              width="14"
            />
            Facebook
          </a>
        </div>
      </body>
    `);
  });
});
