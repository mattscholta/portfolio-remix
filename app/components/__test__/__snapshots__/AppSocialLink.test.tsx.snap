// Vitest Snapshot v1

exports[`AppHero Component > should render our Hero 1`] = `
<body>
  <div>
    <div
      class="hero p-4 text-center"
    >
      <h2
        class="inline-block font-font-serif text-xl font-extrabold md:text-4xl"
      >
        <div
          class="font-font-monospace text-base font-normal md:text-2xl"
        >
          Test Copy
        </div>
        <div
          class="text-highlight px-3 text-4xl md:text-7xl"
        >
          The HIGHLIGHT
        </div>
      </h2>
    </div>
  </div>
</body>
`;

exports[`AppSocialLink Component > should render our Hero 1`] = `
<body>
  <div>
    <a
      class="flex items-center gap-4 break-words break-all text-sm text-color-copy-light"
      href="https://facebook.com"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="Follow me on xxxxx"
        height="14"
        src="facebook"
        width="14"
      />
      Facebook
    </a>
  </div>
</body>
`;
